src/basic/strv.c

```diff
@@ -564,19 +564,19 @@ int strv_extend_front(char ***l, const char *value) {
 
         /* Like strv_extend(), but prepends rather than appends the new entry */
 
-        if (!value)
-                return 0;
-
         n = strv_length(*l);
 
         /* Increase and overflow check. */
         m = n + 2;
         if (m < n)
                 return -ENOMEM;
 
-        v = strdup(value);
-        if (!v)
-                return -ENOMEM;
+        if (value) {
```

Comment:
Shouldn't this commit be reverted ? I personally think this is kind of dangerous to allow this kind of operation, we may leak element...